### ISSUE TITLE
MsgEditValidator not possible

### ISSUE THREAD

**Issue by keupsonite** [cosmos/cosmjs#1299]
The use of the message `MsgEditValidator` is not possible, here is the error received:

```
Log: signature verification failed; please verify account number (XX), sequence (XX) and chain-id (XX): unauthorized
    at SigningStargateClient.broadcastTx (stargateclient.js?d2d5:265:1)
```

No problem with other types of messages. It is possible to do the test directly via https://ezstaking.tools/wallet#validator.

Thanks in advance

**Comment by webmaster128** [comment]
- Which sign node are you using?
- Did you upgrade to CosmJS 0.29.2?
- Did you follow those examples? https://github.com/cosmos/cosmjs/blob/v0.29.2/packages/stargate/src/modules/staking/messages.spec.ts#L147-L288

**Comment by keupsonite** [comment]
Thanks for answering me @webmaster128.

> * Which sign node are you using?
What do you mean? I don't know what you mean.

> * Did you upgrade to CosmJS 0.29.2?
I only updated yesterday from 0.29.1 to 0.29.2, I will try again.

> * Did you follow those examples https://github.com/cosmos/cosmjs/blob/v0.29.2/packages/stargate/src/modules/staking/messages.spec.ts#L147-L288
I did not follow this example, but I have the same thing.

**Comment by keupsonite** [comment]
Ok, it works with version 0.29.2! Thanks a lot @webmaster128.

However, if I don't increase the value of min-self-delegation I get an error message telling me that it's not possible to reduce it, even though I haven't modified it.

**Comment by webmaster128** [comment]
> if I don't increase the value of min-self-delegation I get an error message telling me that it's not possible to reduce it, even though I haven't modified it.

This is a tricky one. In theory there should be a way to not set the fields `commission_rate` and `min_self_delegation` at all and exclude them from the update. Unfortunately I don't know how to encode this in the protobuf API.

**Comment by webmaster128** [comment]
Could you try setting both commission_rate and min_self_delegation to empty strings? That might be the way to not update them.

**Comment by webmaster128** [comment]
I wrote a test myself. Turns out it works for sign mode direct but not Amino JSON. https://github.com/cosmos/cosmjs/pull/1304 adds test cases for that and fixes the Amino JSON encoding.

**Comment by keupsonite** [comment]
Sorry I just saw your notification @webmaster128  I had already done the test with a validator managed via Ledger and it didn't work, I had even tried to define undefined/null. But obviously it was at the level of the toAmino which converted any value received into a string. Thanks a lot for this correction 👐

**Comment by webmaster128** [comment]
Fix should be available in version 0.29.3 now
